level/level.c

```diff
@@ -329,73 +333,116 @@ int rec_partition(int **room_map, int x, int y, int w, int h, int rm) {
 }
 
 static void partition(level *lvl) {
-    int **partitioning = malloc(lvl->width * sizeof(int*));
-    for (int i = 0; i != lvl->width; i++) partitioning[i] = malloc(lvl->height*sizeof(int));
+    // tile-to-room ID mapping
+    //TODO even clearer var name to remove this comment?
+    int **room_ids = malloc(lvl->width * sizeof(int*));
+    for (int i = 0; i != lvl->width; i++) {
+        room_ids[i] = malloc(lvl->height*sizeof(int));
+    }
+
     int **potential_doors= malloc(lvl->width * sizeof(int*));
-    for (int i = 0; i != lvl->width; i++) potential_doors[i] = malloc(lvl->height*sizeof(int));
+    for (int i = 0; i != lvl->width; i++) {
+        potential_doors[i] = malloc(lvl->height*sizeof(int));
+    }
 
-    int rooms = rec_partition(partitioning, 0, 0, lvl->width, lvl->height, 0);
+    // perform partitioning and get highest room ID
+    //TODO clearer function name to remove this comment
+    int max_room_id = rec_partition(room_ids, 0, 0, lvl->width, lvl->height, 0);
 
+    // initialize all tiles to bare floor
+    //TODO Optimization vs. readability: bare floor initialization
+    //TODO could be inside an 'else' down below.
```

Comment:
Thoughts regarding the proposed optimization?

Reply:
Don't care. This could only ever be noticeable on very large maps in which case we've already fallen over for other reasons. Unless for some reason we wanted to create a lot of levels very quickly but outside of testing or weird AI stuff I can't see why that would be interesting.